resources/migrations/058_update_migrations.yaml

```diff
@@ -835,6 +835,77 @@ databaseChangeLog:
             tableName: segment
             columnName: dependency_analysis_version
 
+  - changeSet:
+      id: v58.2025-11-25T17:41:50
+      author: metamben
+      comment: Add collection_id column to metabase_table for publishing tables to collections
+      changes:
+        - addColumn:
+            tableName: metabase_table
+            columns:
+              - column:
+                  name: collection_id
+                  type: int
+                  remarks: The collection this table is published to (null if not published or published to root)
+                  constraints:
+                    nullable: true
+      rollback:
+        - dropColumn:
+            tableName: metabase_table
+            columnName: collection_id
+
+  - changeSet:
+      id: v58.2025-11-25T17:41:51
+      author: metamben
+      comment: Add index on metabase_table.collection_id
+      changes:
+        - createIndex:
+            tableName: metabase_table
+            indexName: idx_metabase_table_collection_id
+            columns:
+              - column:
+                  name: collection_id
+      rollback:
+        - dropIndex:
+            tableName: metabase_table
+            indexName: idx_metabase_table_collection_id
+
+  - changeSet:
+      id: v58.2025-11-25T17:41:52
+      author: metamben
+      comment: Add foreign key constraint from metabase_table.collection_id to collection.id
+      changes:
+        - addForeignKeyConstraint:
```

Comment:
Don't you want to make this `ON DELETE CASCADE`? If not you can just do the FK constrain inline in the `addColumn` migration

Reply:
No, we don't want to cascade. I'm not inlining them because of MySQL. I prefer adding the index first, constraint second so that MySQL doesn't need special handling with its autoindexing FKs.

Reply:
Sorry I meant `ON DELETE SET NULL` -- you delete the collection, set the Table `collection_id` to `NULL` -- but maybe you don't want to do that because of the `is_published` column. If we were only using PG I would say you could write a trigger to unset that as well but I guess it's not super easy to do across three DBs

Only reason I'm asking is because it would make tests easier to write I think

Reply:
Setting to null is dangerous, because that means the table is published in the root collection.